### PR TITLE
[FLOC-2456] Some news entries and documentation for 1.0.1

### PR DESCRIPTION
**Pull request by itamarst**
I have not actually previewed the result, please review carefully!

### CODE REVIEW

File: NEWS [hunk 1]
```diff
@@ -1,3 +1,9 @@
+Flocker 1.0.1 (2015-06-XX)
```
Comment (myechuri) [inline NEWS]: Date needs to be changed from "2015-06-XX" to "2015-06-19" before merging.

File: docs/releasenotes/index.rst [hunk 2]
```diff
@@ -9,6 +9,14 @@ You can learn more about where we might be going with future releases by:
 * Stopping by the ``#clusterhq`` channel on ``irc.freenode.net``.
 * Visiting our GitHub repository at https://github.com/ClusterHQ/flocker.
 
+v1.0.1
+======
+
+Upgrading is strongly recommended for all users of 1.0.0.
```
Comment (davidoliverSP2) [inline docs/releasenotes/index.rst]: This is generally a question for the whole of the docs - do we refer to v1 as v1.0 or v1.0.0 - we use both on this page. 

Also, I would recommend always using the "v"

File: docs/releasenotes/index.rst [hunk 3]
```diff
@@ -9,6 +9,14 @@ You can learn more about where we might be going with future releases by:
...
+
+* The EBS storage driver now more reliably selects the correct OS device file corresponding to an EBS volume being used.
```
Comment (davidoliverSP2) [inline docs/releasenotes/index.rst]: this sentence reads a bit awkwardly - maybe we can reword this to something like this:

The EBS storage driver is now more reliable when selecting the correct OS device file which corresponds with an EBS volume that is being used.

Also, is there a way to define "more reliably"? this is a bit vague.

File: docs/using/administering/upgrading.rst [hunk 4]
```diff
@@ -0,0 +1,34 @@
+=========
+Upgrading
+=========
+
+A Flocker cluster can be upgraded to a newer version of Flocker while preserving all data and configuration of the Flocker cluster provided certain steps are followed.
+The correct steps to follow may vary depending on the versions of Flocker being upgraded from and to.
+A common requirement for all currently supported upgrade paths is that all nodes and the control service must be running the same version of Flocker.
```
Comment (davidoliverSP2) [inline docs/using/administering/upgrading.rst]: This sounds like an important piece of information related to upgrading. Please move this down a line so it starts a new paragraph.

File: docs/using/administering/upgrading.rst [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+
+Flocker 1.0.1
```
Comment (davidoliverSP2) [inline docs/using/administering/upgrading.rst]: As previously mentioned, my recommendation would to use Flocker v1.0.1 and Flocker v1.0.0 when referring to the product version. @ferrantim - do you have a view on this?

File: docs/using/administering/upgrading.rst [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+-------------
+
+Recommended Steps
+^^^^^^^^^^^^^^^^^
+
+#. Stop the agent services on all nodes, and then stop control service.
+#. Install Flocker 1.0.1 on all nodes in the Flocker cluster.
+#. Restart the control service.
+#. If you are using the EBS storage backend reboot each of the agent nodes.
```
Comment (davidoliverSP2) [inline docs/using/administering/upgrading.rst]: missing comma:
If you are using the EBS storage backend, reboot each of the agent nodes.

File: docs/using/administering/upgrading.rst [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+#. If you have not configured the Flocker agents to start automatically on boot,
+   restart the agent services on all nodes.
+
+Details
+^^^^^^^
+
+The upgrade to Flocker 1.0.1 involves changing the way the EBS storage backend maps volumes to devices: in version 1.0.0, there were occasional errors in this mapping.
+As a result, some devices may be mounted in the wrong location.
```
Comment (davidoliverSP2) [inline docs/using/administering/upgrading.rst]: Small change:
As a result, some devices may have been mounted in the wrong location.

**Comment by davidoliverSP2** [comment]
Thanks @itamarst - only a few small recommendations from me, although we probably want a consistent way to refer to the version to use throughout the docs. Hopefully @ferrantim can provide some guidance for us.

File: docs/releasenotes/index.rst (hunk 2 above)
Comment (wallrj) [inline docs/releasenotes/index.rst]: Done.

File: docs/releasenotes/index.rst (hunk 3 above)
Comment (wallrj) [inline docs/releasenotes/index.rst]: I'll leave it I think.

File: docs/using/administering/upgrading.rst (hunk 4 above)
Comment (wallrj) [inline docs/using/administering/upgrading.rst]: I've made it into a note.

File: docs/using/administering/upgrading.rst (hunk 5 above)
Comment (wallrj) [inline docs/using/administering/upgrading.rst]: Changed to v1.0.1

File: docs/using/administering/upgrading.rst (hunk 6 above)
Comment (wallrj) [inline docs/using/administering/upgrading.rst]: Done.

File: docs/using/administering/upgrading.rst (hunk 7 above)
Comment (wallrj) [inline docs/using/administering/upgrading.rst]: Done.

**Comment by wallrj** [comment]
Looks good. 

I addressed most of @davidoliverSP2 's comments; but not all.

I'll merge as soon as the docs builder completes.
